### PR TITLE
[BEAM-3951] Migrate container instructions and builds to Gradle

### PR DESCRIPTION
**Pull request by herohde**
* Also move out Go examples for build performance

### CODE REVIEW

**Comment by herohde** [comment]
R: @lukecwik

File: build_rules.gradle [hunk 1]
```diff
@@ -489,6 +489,22 @@ ext.applyDockerNature = {
   }
 }
 
+def containerImageName(name) {
```
Comment (lukecwik) [inline build_rules.gradle]: Add method comment.

Interesting that def works, before I was sticking everything onto ext. This seems much cleaner. Consider swapping all usages of ext.fooBar = {} to def fooBar

File: sdks/CONTAINERS.md [hunk 2]
```diff
@@ -120,42 +128,45 @@ For the Python SDK harness container image, run:
...
+ff57bdb79ac8: Pushed 
+6e5e20cbf4a7: Pushed 
+86985c679800: Pushed 
+8fad67424c4e: Pushed 
+latest: digest: sha256:86ad57055324457c3ea950f914721c596c7fa261c216efb881d0ca0bb8457535 size: 2646
 ```
 
-Similarly for the Java SDK harness container image. If you want to push the same image
+Similarly for the Java and Go SDK harness container images. If you want to push the same image
 to multiple registries, you can retagging the image using `docker tag` and push.
```
Comment (lukecwik) [inline sdks/CONTAINERS.md]: `retagging` -> `retag`

File: sdks/go/container/build.gradle [hunk 3]
```diff
@@ -46,13 +46,7 @@ golang {
...
 docker {
-  String repositoryRoot
-  if (rootProject.hasProperty(["docker-repository-root"])) {
-    repositoryRoot = rootProject["docker-repository-root"]
-  } else {
-    repositoryRoot = "${System.properties["user.name"]}-docker-apache.bintray.io/beam"
-  }
-  name "${repositoryRoot}/go:latest"
+  name containerImageName("go")
   files "./build/"
```
Comment (lukecwik) [inline sdks/go/container/build.gradle]: Note that with this setup we are building the boot binary multiple times, once per container.

Comment (herohde) [reply sdks/go/container/build.gradle]: Not sure I understand what you mean. We are only building one container image, like before.

File: sdks/CONTAINERS.md (hunk 2 above)
Comment (herohde) [reply sdks/CONTAINERS.md]: Done

File: sdks/go/container/build.gradle (hunk 3 above)
Comment (lukecwik) [reply sdks/go/container/build.gradle]: We have the same golang definition to build the boot.go binary in sdks/java/container, sdks/go/container, and sdks/python/container.

File: build_rules.gradle (hunk 1 above)
Comment (herohde) [reply build_rules.gradle]: Added. I also stick it into ext below.

File: sdks/go/container/build.gradle (hunk 3 above)
Comment (herohde) [reply sdks/go/container/build.gradle]: They are different boot.go files.

**Comment by herohde** [comment]
Thanks @lukecwik. PTAL

**Comment by herohde** [comment]
Updated the helper function. @lukecwik PTAL

**Pull request merged**